src/adapter/graphql/src/scalars/flow_configuration.rs

```diff
@@ -95,42 +95,40 @@ pub enum TimeUnit {
 impl From<chrono::Duration> for TimeDelta {
     fn from(value: chrono::Duration) -> Self {
         let num_weeks = value.num_weeks();
-        if num_weeks > 0 {
-            assert!((value - chrono::Duration::weeks(num_weeks)).is_zero());
+        if (value - chrono::Duration::weeks(num_weeks)).is_zero() {
             return Self {
                 every: num_weeks as u32,
                 unit: TimeUnit::Weeks,
             };
         }
 
         let num_days = value.num_days();
-        if num_days > 0 {
-            assert!((value - chrono::Duration::days(num_days)).is_zero());
+        if (value - chrono::Duration::days(num_days)).is_zero() {
             return Self {
                 every: num_days as u32,
                 unit: TimeUnit::Days,
             };
         }
 
         let num_hours = value.num_hours();
-        if num_hours > 0 {
-            assert!((value - chrono::Duration::hours(num_hours)).is_zero());
+        if (value - chrono::Duration::hours(num_hours)).is_zero() {
             return Self {
                 every: num_hours as u32,
                 unit: TimeUnit::Hours,
             };
         }
 
         let num_minutes = value.num_minutes();
-        if num_minutes > 0 {
-            assert!((value - chrono::Duration::minutes(num_minutes)).is_zero());
+        if (value - chrono::Duration::minutes(num_minutes)).is_zero() {
             return Self {
                 every: num_minutes as u32,
                 unit: TimeUnit::Minutes,
             };
         }
 
-        unreachable!("Expecting intervals not tinier than 1 minute");
+        unreachable!(
```

Comment:
`unreachable` is usually for things you are certain won't happen. If our `From` can fail on a whole bunch of `Duration` values perhaps it should be a `TryFrom` that returns an error rather than panics the whole program?

Reply:
I'm certain the things won't happen, as, assuming correct implementation, API only returns values that were previously set via API, and those cannot include mixed unit itervals like 1h 25 min